src/moscot/problems/cross_modality/_mixins.py

```diff
@@ -0,0 +1,226 @@
+            # Projecting the source domain onto the target domain
+            self._tgt_aligned = self.adata_tgt.obsm[self._tgt_attr]
+            self.coupling = self[('src', 'tgt')].solution.transport_matrix
+            weights = np.sum(self.coupling, axis = 1)
+            self._src_aligned = np.matmul(self.coupling, self._tgt_aligned) / weights[:, None]
+        else:
+            # Projecting the target domain onto the source domain
+            self._src_aligned = self.adata_src.obsm[self._src_attr]
+            self.coupling = self[('src', 'tgt')].solution.transport_matrix
+            weights = np.sum(self.coupling, axis = 1)
+            self._tgt_aligned = np.matmul(np.transpose(self.coupling), self._src_aligned) / weights[:, None]
+        
+        self.adata_src.obsm["X_aligned"] = self._src_aligned
+        self.adata_tgt.obsm["X_aligned"] = self._tgt_aligned
+        return self._src_aligned, self._tgt_aligned
+    
+    def integrate(
+            self: CrossModalityIntegrationMixinProtocol[K, B],
+            normalize = True,
+            norm = "l2",
+            SRContoTGT=True,
+            **kwargs:Any,
+    ) -> ArrayLike:
+        """
+        Integrate source and target objects.
+
+        Parameters
+        ----------
+        %(normalize)s
+        %(norm)s
+        %(SRContoTGT)s
+
+        Returns
+        -------
+        
+        """
+        if normalize:
+            self._normalize(norm=norm) # überschreibt so die adata objecte, evlt. lieber neues feld in obsm hinzufügen?
+        
+        src_aligned, tgt_aligned = self._barycentric_projection(SRContoTGT=SRContoTGT)
+
+        self.src_aligned, self.tgt_aligned = src_aligned, tgt_aligned
+        return (self.src_aligned, self.tgt_aligned)
+                        
+    def foscttm(
+            self: CrossModalityIntegrationMixinProtocol[K, B],
+            **kwargs,
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """
+        Fraction of samples closer than true match (smaller is better)
+        Parameters
+        ----------
+        **kwargs
+            Additional keyword arguments are passed to
+            :func:`scipy.spatial.distance_matrix`
+        Returns
+        -------
+        fracs: FOSCTTM for samples in source and target modality
+        
+        Note
+        ----
+        Samples in source and target modality should be paired and given in the same order
+        """
+    
+        x = self.adata_src.obsm["X_aligned"] 
+        y = self.adata_tgt.obsm["X_aligned"] 
+        if x.shape != y.shape:
+            raise ValueError("Shapes do not match!")
+        d = scipy.spatial.distance_matrix(x, y, **kwargs)
+        foscttm_x = (d < np.expand_dims(np.diag(d), axis=1)).mean(axis=1)
+        foscttm_y = (d < np.expand_dims(np.diag(d), axis=0)).mean(axis=0)
+        fracs = []
+        for i in range(len(foscttm_x)):
+            fracs.append((foscttm_x[i]+foscttm_y[i])/2)
+        return np.mean(fracs).round(4)    
```

Comment:
we can remove this because we do not always have the ground truth. this is for utiliy, not benchmark.